singleheader/simdjson.cpp

```diff
@@ -5771,10 +5770,31 @@ struct value128 {
   uint64_t high;
 };
 
+#if defined(_MSC_VER) && !defined(_M_X64) // _umul128 for x86, arm, arm64
+#if defined(_M_ARM)
+static inline uint64_t __emulu(uint32_t x, uint32_t y) {
+  return x * (uint64_t)y;
+}
+#endif
+static inline uint64_t _umul128(uint64_t ab, uint64_t cd, uint64_t *hi) {
```

Comment:
For ARM 64-bits, I think that this should be _umul128 instead of this expensive 32-bit emulation.

https://github.com/miranda-ng/miranda-ng/blob/9aa5329a4b4077970571f40f79486a3a1bf3e9f8/libs/libmdbx/src/test/utils.h#L53

Reply:
Oh. I see that you are ahead of me. Good.

Reply:
Oh. Wait... No.